eth/api.go

```diff
@@ -452,10 +444,12 @@ func formatError(err error) string {
 
 // TraceTransaction returns the structured logs created during the execution of EVM
 // and returns them as a JSON object.
-func (api *PrivateDebugAPI) TraceTransaction(txHash common.Hash, logger *vm.LogConfig) (*ethapi.ExecutionResult, error) {
-	if logger == nil {
-		logger = new(vm.LogConfig)
+func (api *PrivateDebugAPI) TraceTransaction(txHash common.Hash, logConfig *vm.LogConfig) (*ethapi.ExecutionResult, error) {
+	if logConfig == nil {
+		logConfig = new(vm.LogConfig)
 	}
+	logger := vm.NewLogger(*logConfig)
```

Comment:
`NewLogger` takes a value, not a pointer, so we have to have something to dereference safely.